lib/Optimizer/Transforms/ArgumentSynthesis.cpp

```diff
@@ -31,121 +31,120 @@ class ArgumentSynthesisPass
   using ArgumentSynthesisBase::ArgumentSynthesisBase;
 
   void runOnOperation() override {
-    func::FuncOp func = getOperation();
-    StringRef funcName = func.getName();
-    std::string text;
-    if (std::find_if(funcList.begin(), funcList.end(),
-                     [&](const std::string &item) {
-                       auto pos = item.find(':');
-                       if (pos == std::string::npos)
-                         return false;
-                       std::string itemName = item.substr(0, pos);
-                       bool result = itemName == funcName;
-                       if (result)
-                         text = item.substr(pos + 1);
-                       return result;
-                     }) == funcList.end()) {
-      // If the function isn't on the list, do nothing.
-      LLVM_DEBUG(llvm::dbgs() << funcName << " not in list.\n");
-      return;
-    }
+    ModuleOp moduleOp = getOperation();
+    for (auto item : funcList) {
+      auto pos = item.find(':');
+      if (pos == std::string::npos)
+        continue;
 
-    // If there are no substitutions, we're done.
-    if (text.empty()) {
-      LLVM_DEBUG(llvm::dbgs() << funcName << " has no substitutions.");
-      return;
-    }
+      std::string funcName = item.substr(0, pos);
```

Comment:
```suggestion
      StringRef funcName = item.substr(0, pos);
```
Don't think we need to make a copy of this string?

Reply:
This does not compile - a temp object is destroyed.